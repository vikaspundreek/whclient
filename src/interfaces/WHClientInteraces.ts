export interface IWHConfig {
  apiEndPoint: string;
  tabs: IWHClientTab[];
}
export interface IWHClientTab {
  id: string;
  label: string;
}

export interface IProduct {
  id: string;
  name: string;
  articles: IArticle[];
  stock?: number;
}

export interface IArticle {
  id: string;
  name?: string;
  amountInStock?: number;
  amountRequired: number;
}

export interface ISale {
  id: string;
  createdAt: string;
  productId: string;
  productName?: string;
  amountSold: number;
}

export interface IStore {
  products: IProduct[];
  articles: IArticle[];
  sales: ISale[] | null;
}

export interface IBulkArticleUpdate {
  id: string;
  amountToSubtract: number;
}
