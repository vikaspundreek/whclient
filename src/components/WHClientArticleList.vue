<template>
  <table class="table">
    <thead>
      <tr>
        <th scope="col">Article Name</th>
        <th scope="col" class="text-right">Required</th>
        <th scope="col" class="text-right">Availability</th>
      </tr>
    </thead>
    <tbody>
      <tr
        class="wh-client__article pt-2"
        v-for="article in articles"
        :key="article.index"
      >
        <td>{{ article.name }}</td>
        <td class="text-right">{{ article.amountRequired }}</td>
        <td
          class="text-right wh-client__stock"
          v-html="getAvailability(article)"
        />
      </tr>
    </tbody>
  </table>
</template>
<script lang="ts">
import { Component, Prop, Vue } from "vue-property-decorator";
import { IArticle } from "../interfaces/WHClientInteraces";

@Component
export default class WHClientArticleList extends Vue {
  @Prop() private articles!: IArticle[];
  private getAvailability(article: IArticle) {
    return article.amountInStock &&
      article.amountInStock >= article.amountRequired
      ? `<span class="article-stock">${article.amountInStock}</></span>`
      : '<span class="article-stock article-stock--unavailable">N/A</span>';
  }
}
</script>
