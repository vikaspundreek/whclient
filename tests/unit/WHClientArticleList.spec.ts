import { shallowMount } from "@vue/test-utils";
import WHClientArticleList from "../../src/components/WHClientArticleList.vue";
import { IArticle } from "@/interfaces/WHClientInteraces";
const mockArticleList: IArticle[] = [
  {
    id: "0517f083-0e15-4876-8d1f-6fa45900431c",
    amountRequired: 4,
    name: "Leg",
    amountInStock: 12,
  },
  {
    id: "831b92b8-677b-42cc-a585-335ea4ccccb6",
    amountRequired: 1,
    name: "Seat",
    amountInStock: 2,
  },
  {
    id: "addc65a8-c759-41d8-a18a-89fe446ad484",
    amountRequired: 8,
    name: "Screw",
    amountInStock: 17,
  },
];

describe("WHClientArticleList.vue", () => {
  it("Render article list as per provided article data", () => {
    const wrapper = shallowMount(WHClientArticleList, {
      propsData: { articles: mockArticleList },
    });
    expect(wrapper).toMatchSnapshot();
  });
  it("Render article list with one article is availability less than requirement", () => {
    mockArticleList[0].amountInStock = mockArticleList[0].amountRequired - 1;
    const wrapper = shallowMount(WHClientArticleList, {
      propsData: { articles: mockArticleList },
    });
    expect(wrapper).toMatchSnapshot();
  });
});
