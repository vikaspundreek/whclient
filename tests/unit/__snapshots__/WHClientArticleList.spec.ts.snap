// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WHClientArticleList.vue Render article list as per provided article data 1`] = `
<table class="table">
  <thead>
    <tr>
      <th scope="col">Article Name</th>
      <th scope="col" class="text-right">Required</th>
      <th scope="col" class="text-right">Availability</th>
    </tr>
  </thead>
  <tbody>
    <tr class="wh-client__article pt-2">
      <td>Leg</td>
      <td class="text-right">4</td>
      <td class="text-right wh-client__stock"><span class="article-stock">12</span></td>
    </tr>
    <tr class="wh-client__article pt-2">
      <td>Seat</td>
      <td class="text-right">1</td>
      <td class="text-right wh-client__stock"><span class="article-stock">2</span></td>
    </tr>
    <tr class="wh-client__article pt-2">
      <td>Screw</td>
      <td class="text-right">8</td>
      <td class="text-right wh-client__stock"><span class="article-stock">17</span></td>
    </tr>
  </tbody>
</table>
`;

exports[`WHClientArticleList.vue Render article list with one article is availability less than requirement 1`] = `
<table class="table">
  <thead>
    <tr>
      <th scope="col">Article Name</th>
      <th scope="col" class="text-right">Required</th>
      <th scope="col" class="text-right">Availability</th>
    </tr>
  </thead>
  <tbody>
    <tr class="wh-client__article pt-2">
      <td>Leg</td>
      <td class="text-right">4</td>
      <td class="text-right wh-client__stock"><span class="article-stock article-stock--unavailable">N/A</span></td>
    </tr>
    <tr class="wh-client__article pt-2">
      <td>Seat</td>
      <td class="text-right">1</td>
      <td class="text-right wh-client__stock"><span class="article-stock">2</span></td>
    </tr>
    <tr class="wh-client__article pt-2">
      <td>Screw</td>
      <td class="text-right">8</td>
      <td class="text-right wh-client__stock"><span class="article-stock">17</span></td>
    </tr>
  </tbody>
</table>
`;
